src/gg.regression.unity.bots/Editor/Scripts/ActionManager/RGActionAnalysis.cs

```diff
@@ -1126,9 +1126,80 @@ private void AnalyzeGameObject(GameObject gameObject)
             {
                 foreach (string listener in RGActionManagerUtils.GetEventListenerMethodNames(btn.onClick))
                 {
-                    _buttonClickListeners.Add(listener);
+                    string[] path = {"Unity UI", "Button", listener};
+                    AddAction(new UIButtonPressAction(path, typeof(Button), listener), null);
                 }
             }
+            if (gameObject.TryGetComponent(out Toggle _) && !IsRGOverlayObject(gameObject))
```

Comment:
With the current structure the RGOverlay test only happens if the game object actually contains a UI component. I figure this would be more efficient than doing the overlay test on every game object, which traverses the parents and checks the object names. Though I am open to feedback on better ways to detect if the object is part of the RG overlay canvas.